gitlab/client.py

```diff
@@ -653,7 +653,7 @@ def http_request(
         retry_transient_errors: Optional[bool] = None,
         max_retries: int = 10,
         **kwargs: Any,
-    ) -> requests.Response:
+    ) -> _backends.DefaultResponse:
```

Comment:
I'll wait for https://github.com/python-gitlab/python-gitlab/pull/2442 to get merged and I think we can then do some magic with generic types perhaps.:mage:

Reply:
The end goal is to move the usage of the requests package to the backend. Using generic types like Union, can work for a transition period, but eventually, it will have to be removed.

Reply:
I had in mind more something to do with `TypeVar` to define acceptable types or an upper bound, so we could include both requests and httpx. But this works for me for now.